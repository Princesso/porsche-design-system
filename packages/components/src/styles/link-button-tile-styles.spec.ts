import {
  getButtonLinkTileContentStyles,
  getButtonLinkTilePStyles,
  getButtonLinkTileSharedStyles,
} from './link-button-tile-styles';

describe('getButtonLinkTilePStyles()', () => {
  it.each<Parameters<typeof getButtonLinkTilePStyles>>([
    ['default', 'semi-bold'],
    ['inherit', 'semibold'],
    ['default', 'regular'],
    [
      { base: 'inherit', xs: 'default', s: 'inherit', m: 'default', l: 'inherit', xl: 'default' },
      { base: 'semi-bold', xs: 'regular', s: 'semibold', m: 'regular', l: 'semi-bold', xl: 'regular' },
    ],
  ])('should return correct css for %o', (...args) => {
    expect(getButtonLinkTilePStyles(...args)).toMatchSnapshot();
  });
});

describe('getButtonLinkTileContentStyles()', () => {
  it.each<Parameters<typeof getButtonLinkTileContentStyles>>([
    ['top', true, false],
    ['bottom', false, true],
    ['top', false, { base: false, xs: true, s: false, m: true, l: false, xl: true }],
  ])('should return correct css for %o', (...args) => {
    expect(getButtonLinkTileContentStyles(...args)).toMatchSnapshot();
  });
});

describe('getButtonLinkTileSharedStyles()', () => {
  it.each<Parameters<typeof getButtonLinkTileSharedStyles>>([
    [true],
    [false],
    [{ base: false, xs: true, s: false, m: true, l: false, xl: true }],
  ])('should return correct css for %o', (...args) => {
    expect(getButtonLinkTileSharedStyles(...args)).toMatchSnapshot();
  });
});
