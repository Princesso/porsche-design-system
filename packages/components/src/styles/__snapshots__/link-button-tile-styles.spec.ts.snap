// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getButtonLinkTileContentStyles() should return correct css for 'bottom' 1`] = `
{
  "alignItems": "center",
  "bottom": 0,
  "display": "grid",
  "gridTemplateColumns": "auto 24px",
  "gridTemplateRows": "auto",
  "padding": "clamp(32px, 2.75vw + 23px, 76px) clamp(16px, 1.25vw + 12px, 36px) clamp(16px, 1.25vw + 12px, 36px)",
}
`;

exports[`getButtonLinkTileContentStyles() should return correct css for 'top' 1`] = `
{
  "background": "linear-gradient(to top, rgba(31,31,31,0.9) 0%,rgba(31,31,31,0.9) 20%,rgba(31,31,31,0.852589) 26.67%,rgba(32,32,32,0.768225) 33.33%,rgba(33,33,33,0.668116) 40%,rgba(34,34,34,0.557309) 46.67%,rgba(35,35,35,0.442691) 53.33%,rgba(36,36,36,0.331884) 60%,rgba(37,37,37,0.231775) 66.67%,rgba(38,38,38,0.147411) 73.33%,rgba(39,39,39,0.0816599) 80%,rgba(39,39,39,0.03551) 86.67%,rgba(39,39,39,0.0086472) 93.33%,rgba(39,39,39,0) 100%);",
  "display": "grid",
  "gridTemplateColumns": "auto",
  "gridTemplateRows": "auto auto",
  "padding": "clamp(16px, 1.25vw + 12px, 36px) clamp(16px, 1.25vw + 12px, 36px) clamp(32px, 2.75vw + 23px, 76px)",
  "top": 0,
}
`;

exports[`getButtonLinkTileContentStyles() should return correct css for 'top' 2`] = `
{
  "@media(min-width:1000px)": {
    "alignItems": "center",
    "gridTemplateColumns": "auto 24px",
    "gridTemplateRows": "auto",
    "top": 0,
  },
  "@media(min-width:1300px)": {
    "gridTemplateColumns": "auto",
    "gridTemplateRows": "auto auto",
  },
  "@media(min-width:1760px)": {
    "alignItems": "center",
    "gridTemplateColumns": "auto 24px",
    "gridTemplateRows": "auto",
    "top": 0,
  },
  "@media(min-width:480px)": {
    "alignItems": "center",
    "gridTemplateColumns": "auto 24px",
    "gridTemplateRows": "auto",
    "top": 0,
  },
  "@media(min-width:760px)": {
    "gridTemplateColumns": "auto",
    "gridTemplateRows": "auto auto",
  },
  "display": "grid",
  "gridTemplateColumns": "auto",
  "gridTemplateRows": "auto auto",
  "padding": "clamp(16px, 1.25vw + 12px, 36px) clamp(16px, 1.25vw + 12px, 36px) clamp(32px, 2.75vw + 23px, 76px)",
  "top": 0,
}
`;

exports[`getButtonLinkTilePStyles() should return correct css for 'default' 1`] = `
{
  "p": {
    "color": "#FBFCFF",
    "font": "normal normal 400 clamp(1.27rem, 0.51vw + 1.16rem, 1.78rem)/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
    "fontSize": "clamp(1.13rem, 0.21vw + 1.08rem, 1.33rem)",
    "fontWeight": 600,
    "hyphens": "auto",
    "margin": 0,
    "maxWidth": "34.375rem",
    "overflowWrap": "break-word",
  },
}
`;

exports[`getButtonLinkTilePStyles() should return correct css for 'default' 2`] = `
{
  "p": {
    "color": "#FBFCFF",
    "font": "normal normal 400 clamp(1.27rem, 0.51vw + 1.16rem, 1.78rem)/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
    "fontSize": "clamp(1.13rem, 0.21vw + 1.08rem, 1.33rem)",
    "fontWeight": 400,
    "hyphens": "auto",
    "margin": 0,
    "maxWidth": "34.375rem",
    "overflowWrap": "break-word",
  },
}
`;

exports[`getButtonLinkTilePStyles() should return correct css for 'inherit' 1`] = `
{
  "p": {
    "color": "#FBFCFF",
    "font": "normal normal 400 clamp(1.27rem, 0.51vw + 1.16rem, 1.78rem)/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
    "fontSize": "inherit",
    "fontWeight": 600,
    "hyphens": "auto",
    "margin": 0,
    "maxWidth": "34.375rem",
    "overflowWrap": "break-word",
  },
}
`;

exports[`getButtonLinkTilePStyles() should return correct css for {
  base: 'inherit',
  xs: 'default',
  s: 'inherit',
  m: 'default',
  l: 'inherit',
  xl: 'default'
} 1`] = `
{
  "p": {
    "@media(min-width:1000px)": {
      "fontSize": "clamp(1.13rem, 0.21vw + 1.08rem, 1.33rem)",
      "fontWeight": 400,
    },
    "@media(min-width:1300px)": {
      "fontSize": "inherit",
      "fontWeight": 600,
    },
    "@media(min-width:1760px)": {
      "fontSize": "clamp(1.13rem, 0.21vw + 1.08rem, 1.33rem)",
      "fontWeight": 400,
    },
    "@media(min-width:480px)": {
      "fontSize": "clamp(1.13rem, 0.21vw + 1.08rem, 1.33rem)",
      "fontWeight": 400,
    },
    "@media(min-width:760px)": {
      "fontSize": "inherit",
      "fontWeight": 600,
    },
    "color": "#FBFCFF",
    "font": "normal normal 400 clamp(1.27rem, 0.51vw + 1.16rem, 1.78rem)/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif",
    "fontSize": "inherit",
    "fontWeight": 600,
    "hyphens": "auto",
    "margin": 0,
    "maxWidth": "34.375rem",
    "overflowWrap": "break-word",
  },
}
`;

exports[`getButtonLinkTileSharedStyles() should return correct css for { base: false, xs: true, s: false, m: true, l: false, xl: true } 1`] = `
{
  "link-or-button": {
    "@media(min-width:1000px)": {
      "display": "none",
    },
    "@media(min-width:1300px)": {
      "display": "inline-block",
    },
    "@media(min-width:1760px)": {
      "display": "none",
    },
    "@media(min-width:480px)": {
      "display": "none",
    },
    "@media(min-width:760px)": {
      "display": "inline-block",
    },
    "display": "inline-block",
    "minHeight": "54px",
  },
  "link-or-button-pure": {
    "@media(min-width:1000px)": {
      "display": "inline-block",
    },
    "@media(min-width:1300px)": {
      "display": "none",
    },
    "@media(min-width:1760px)": {
      "display": "inline-block",
    },
    "@media(min-width:480px)": {
      "display": "inline-block",
    },
    "@media(min-width:760px)": {
      "display": "none",
    },
    "display": "none",
  },
}
`;

exports[`getButtonLinkTileSharedStyles() should return correct css for false 1`] = `
{
  "link-or-button": {
    "display": "inline-block",
    "minHeight": "54px",
  },
  "link-or-button-pure": {
    "display": "none",
  },
}
`;

exports[`getButtonLinkTileSharedStyles() should return correct css for true 1`] = `
{
  "link-or-button": {
    "display": "none",
    "minHeight": "54px",
  },
  "link-or-button-pure": {
    "display": "inline-block",
  },
}
`;
